### ISSUE TITLE
Set "source:direction" with new direction-compass-feature

### ISSUE THREAD

**Issue by tordans** [bryceco/GoMap#188]
@wtimme I created a new issue for this based on https://github.com/bryceco/GoMap/pull/180 since IMO this is an add on to your PR, nothing that blocks it.

---

I wonder if it's a good idea to also add a `source:direction=mobile phone compass, GoMap!!` / `source:camera:direction=*`. Similar to what we discussed at https://github.com/bryceco/GoMap/issues/159 whenever the feature is used.

What I want to solve is: Whenever I review a OSM-node/-way while #Mapwalking I want a quick way to validate the data. Something like "direction=192" would look odd, since most directions ATM are rounded numbers (see https://taginfo.openstreetmap.org/keys/direction#values).

The source-key/value would help, to understand how the data was created and also how valid it might be.

I understand, that the Wiki (https://wiki.openstreetmap.org/wiki/Key:source) is more inclined to only use changeset source-tags, but those would IMO not help to glue the source-info to the specific tag that was change (especially with large changesets). Also, there seems to be no clear "rule" against using the source.

What do you think?

**Comment by wtimme** [comment]
Like @zlavergne worte in #159, the [documentation for the key `source`][1] states:

> With modern editors, the source=* tag is typically added to the changeset when a change is made, not as a tag on a piece of information which may be updated at different times based on different sources.

So they advise against using something like `source:direction`, because it's might get outdated at some point.

By looking at an object's versions and their respective changesets, we should be able to determine the `source` tag of the changeset in which the value was last updated, thus giving us the information you are looking for. Relying on the tags of changesets would further allow us to use this with any key, not just `direction`.

[1]: https://wiki.openstreetmap.org/wiki/Key:source

**Comment by bryceco** [comment]
I’m not a fan of adding this tag to objects for similar reasons.

**Comment by tordans** [comment]
Hi! I disagree with the logic of https://github.com/bryceco/GoMap/issues/188#issuecomment-481389349 but I don't think its worth the effort to discuss it further. We have more interesting things to do :-). Will close the other issue as well.

**Comment by wtimme** [comment]
Hi, I still would love to hear from your experience. What is it that you are looking for? Can you maybe elaborate on the usecase? I probably didn't get it right.

**Comment by tordans** [comment]
@wtimme the way I look at it is …

A changeset can have n objects.
A object can have n key.s
A `source`-key on the changeset is for the whole changeset.
A `source`-key on the object is for one object, so more detailed.
A `source:keyname`-key on an object is for one key on that object, so even more detailed.

I can have a changeset with 50 objects. The changeset-source says: "local knowledge, mapwalking".
Part of that is a bench that I added while looking at a map of a park. This object's-source says "map of park by XY".
Another part of this changeset is an existing tree-object that I just modified by adding a height with GoMap. This object's-"source:height"-key says "Calculated by GoMap heights-feature"
Another part of this changeset is a tree-object that I added, but there GPS was bad, so I could not use the GoMap feature, so I eyeballed it based on other data, like the tree from above. I would not add a "source:height" here, since the default for height is IMO "source estimated" (I dont know if this is correct), but if I had to, I would say just that in the source:heigh-text.

The tree in https://overpass-api.de/achavi/?changeset=69050488&zoom=19&lat=41.84644&lon=12.56158&layers=B0TTTFT at https://www.openstreetmap.org/node/6394077609#map=19/41.84625/12.56133 is an example. This one I could measure quite OK with GoMap. Other trees nearby had +/- 8m in accuracy, so I either estimated them based on this tree or I just did not add a height at all.

So to sum it up, all three types of `source` have their place/use; it's just a very different message for the same key.
